cou/steps/analyze.py

```diff
@@ -84,7 +84,7 @@ def is_data_plane(app: OpenStackApplication) -> bool:
         for app in apps:
             if is_data_plane(app):
                 data_plane.append(app)
-            elif any(unit.machine in data_plane_machines for unit in app.units.values()):
+            elif any(machine in data_plane_machines for machine in app.machines.values()):
```

Comment:
What is the difference here?

Reply:
Och, I see, subordinate does not have units. Nice catch.

Reply:
I realized that this is not a proper fix, since we are getting machines from `status.units` [here](https://github.com/canonical/charmed-openstack-upgrader/blob/e00c5824bbe8f8e3ed7e263a16b42855fedced67/cou/utils/juju_utils.py#L358) and that will be empty for even subordinate.
So we need to change getting those machines for app or do something else.